lib/internal/modules/cjs/loader.js

```diff
@@ -85,6 +85,27 @@ const relativeResolveCache = Object.create(null);
 
 let requireDepth = 0;
 let statCache = null;
+
+function enrichCJSError(err) {
+  const stack = err.stack.split('\n');
+
+  const lineWithErr = stack.findIndex((line) =>
+    (/\^{9}/).test(line));
+
+  let previousLine = false;
+
+  if (lineWithErr !== -1) {
+    previousLine = stack[lineWithErr - 1];
+  }
+
+  if (err.message.startsWith('Unexpected token export') ||
+    (previousLine && previousLine.includes('import'))) {
```

Comment:
This will fail for `var importName = 'invalid;`.

To make it more resiliant, we can use the fact that import will always take one of the following forms (if valid syntax):
```
import { ... } from 'x'
import 'x'
import "x"
import x from 'x'
import * as X from 'x'
```

So if we directly test those variations in a regex we can avoid the invalid cases, which we must do.

Reply:
the regexp used now covers most of these cases additionally

```
import {
something
} from "somewhere";
```

but doesn't cover 

```
/* comment */ import "x";
```

i tried to work with the regexp more but i couldn't get it right, any suggestions?